hbase-shell/src/main/ruby/shell.rb

```diff
@@ -335,12 +348,9 @@ def eval_io(io)
         # Include the 'ERROR' string to try to make transition easier for scripts that
         # may have already been relying on grepping output.
         puts "ERROR #{e.class}: #{message}"
-        if $fullBacktrace
-          # re-raising the will include a backtrace and exit.
-          raise e
-        else
-          exit 1
-        end
+        raise e unless hide_traceback
+
+        exit 1
```

Comment:
Control flow returns to this line after a `raise` ? `raise` is an expression in this language..?

Reply:
Ruby allows one-line conditionals such as `<statement> if <condition>`, which is equivalent to `if <condition>; <statement>; end`. It's a fun feature of Ruby :smile:.

So, this logic will raise SystemExit (via `exit 1`) if we want to hide the traceback.

Reply:
I also added unit tests that cover/verify this.